DESIGN.md

```diff
@@ -0,0 +1,77 @@
+# Design document
+
+Felek runs arbitrary Unix processes ("jobs") on the server machine, allowing a client to query job status, stop running jobs, and inspect its stdout and stderr streams ("logs").
+
+## Architecture
+
+Felek uses Go's `exec.Cmd` to run external processes, and provides a gRPC interface for clients.
+
+When a `Start` request is received, Felek will try to start an external process using the provided settings (command path, arguments, environment variables, and a working directory path). If the process was started successfully, Felek will generate a unique ID for it an store it in a key-value store for later queries.
```

Comment:
> will generate a unique ID

How will it be generated?

Reply:
I was thinking of just using [`Int63`](https://golang.org/pkg/math/rand/#Int63) blindly, since I think handling collisions is probably outside of scope. If it's not, we can use `Int63` check for collisions, regenerating until a free ID is found.

Reply:
You could also stick to something like standardized UUIDs to avoid checking for collisions.

Reply:
Can't you represent UUID like a string?

Reply:
Yeah, but that's wasteful, space-wise, but I guess that's fine. Will switch to using a string.